app/com/github/pedrovgs/dotto/interpreter/DottoInterpreter.scala

```diff
@@ -0,0 +1,46 @@
+package com.github.pedrovgs.dotto.interpreter
+
+import akka.actor.ActorRef
+import cats.{Id, ~>}
+import com.github.pedrovgs.dotto.core.algebra.dotto._
+import com.github.pedrovgs.dotto.interpreter.actors.ShowMessageActor.ShowMessage
+import com.github.pedrovgs.dotto.interpreter.led.ledController._
+import com.github.pedrovgs.dotto.interpreter.morse.morseTranslator._
+import play.api.Logger
+
+/**
+  * Production interpreter implemented using Cats as core library.
+  */
+class DottoInterpreter(val showMessageActor: ActorRef) extends (Instruction ~> Id) {
+
+  def apply[A](i: Instruction[A]): Id[A] = i match {
+    case EnqueueMessage(messageToEnqueue) => enqueueMessage(messageToEnqueue)
+    case TranslateMessageIntoMorse(messageToTranslate) => translateMessageIntoMorse(messageToTranslate)
+    case TranslateIntoLedInteractions(morseSentence) => translateIntoLedInteractions(morseSentence)
+    case ShowLedInteractions(ledInteractions) => showMorseSentence(ledInteractions)
+  }
+
+  private def enqueueMessage(message: Message): Message = {
+    Logger.debug("Message enqueued to be translated: " + message)
```

Comment:
While this is common practice in Scala this log calls are actually side effects and make all these functions impure. You may want to look into MonadWriter or similar cats alternatives for pure logging. Alternatively model this effect in your ADT and provide impls for them in the interpreters. I still think there is no point for actors here.